docs(concepts): Update resources and features, add account linking (#7934)
zitadel/zitadel: docs/docs/concepts/structure/users.md

### Federated users

Federated users are identities that are managed by a third-party identity provider.
Users can login via an external identity provider, using [identity brokering](../features/identity-brokering) ("Single-sign-on").
Federated user [accounts are linked](../features/account-linking) to internal users to be able to assign roles and keep an audit trail.

### External users

In a multi-tenancy architecture, you might use [organizations](organizations) to separate user groups.
By using [external user grants](../features/external-user-grant) an organization is able to invite users from another organization.
These invited users are called external users.

## Considerations

### Uniqueness of users

Users can only exist within one [organization](/concepts/structure/organizations).
It is currently not possible to move users between organizations.

User accounts are uniquely identified by their `id` or `loginname` in combination of the `organization domain` (eg, `road.runner@acme.zitadel.local`).
You can use the same email address for different user accounts.

### How to structure user pools

Consider this general recommendation as a starting point:

> - Create one organization ("default organization") for your own team

Author: ```suggestion
- Create one organization ("default organization") for your own company
```

- Configure projects and applications in the default organization

> - Structure users in the smallest unit of self-managed groups ("teams", "user pools")

Reviewer: Do we really recommend this generally? Problem I see when you add teams from a company, you can only have the domain once for an organization.
e.g if you have company-a and the team marketing and engineering, you can put them all in the same or or in different. but only in the same org you will be able to add theverified domain. 
I would not generally recommend adding the teams. but there might be use cases where it makes sense.

Author: @hifabienne sorry with team I meant for the own company. I will re-write that. We call an organization in the cloud also Team... so I wanted to get closer to that. But I see that this is confusing. We should also keep that in mind for the cloud.

Author: I've also added a consideration below. For me this is still the basic recommendation and then from there on the users need to figure out why this would not work in their case (like you just did). We need to provide a better series here anyways. 

FYI - @dakshitha please see comments from Fabi here on the architecture.

> - Structure users in the smallest unit of self-managed groups ("teams", "user pools")

Author: ```suggestion
- Structure users in organizations based on common domains that are self-managed (eg, company)
```

- Grant your projects to the organizations, allow Managers to give granted roles to their users

> You might want to adjust this general setup based on your [scenario](/guides/solution-scenarios/introduction).

Author: ```suggestion
You might want to adjust this general setup based on your [scenario](/guides/solution-scenarios/introduction). 

One important consideration in the setup is that you can only have a domain once for an organization. If you have multiple teams working with the same email address, you might need to add them to one single organization that has the domain verified for the teams' domain.
```
